SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Build/Build.cs

```diff
@@ -89,17 +89,33 @@ public static void Main(string[] args)
             // Locate the Unreal Engine.
             Console.WriteLine("Finding Unreal Engine build.");
             string uproject = File.ReadAllText(projectFile, Encoding.UTF8);
-            string engineAssociationRegex = "..{8}-.{4}-.{4}-.{4}-.{12}.";
-            Match engineAssociationMatch = Regex.Match(uproject, engineAssociationRegex);
 
             string unrealEngine = "";
+            string engineAssociation = "";
 
-            // If we found an engine association within the .uproject, query the registry for the engine directory.
-            if (engineAssociationMatch.Success)
-            {
-                string engineAssociation = engineAssociationMatch.Value;
-                Console.WriteLine("Engine Association: " + engineAssociation);
+            dynamic projectJson = JObject.Parse(uproject);
+            dynamic assocation = projectJson.EngineAssociation;
+            engineAssociation = assocation;
+
+            Console.WriteLine("Engine Association: " + engineAssociation);
 
+            // If the engine association is empty then climb the parent directories of the project looking for the "Engine" and Build.version.
+            if (string.IsNullOrEmpty(engineAssociation))
+            {
+                DirectoryInfo currentDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+                while (currentDir.Parent != null && string.IsNullOrEmpty(unrealEngine))
+                {
+                    currentDir = currentDir.Parent;
+                    unrealEngine = currentDir.GetDirectories().Where(d => d.Name == "Engine" && File.Exists(Path.Combine(d.FullName, @"Build\Build.version"))).Select(d => d.Parent.FullName).FirstOrDefault();
```

Comment:
Why not just build the full path up with all of the information you have and check `File.Exists` on that?
```
var testEngine = Path.Combine(currentDir.FullName, "Engine");
if (File.Exists(testEngine, "Build", "Build.version"))
{
	unrealEngine = testEngine;
	break;
}
```

Reply:
Works too, makes the code bit simpler too.